app/models/spree/order_contents_decorator.rb

```diff
@@ -10,17 +31,34 @@ def add_to_line_item_with_parts(variant, quantity, options = {})
         )
       end
     end
+
     alias_method_chain :add_to_line_item, :parts
 
     private
 
+    def part_variants_match(line_item, variant, quantity, options)
```

Comment:
Considered rename this method to `part_variants_match?` since it returns boolean.